salt/edx/mitx_etl.sls

```diff
@@ -0,0 +1,57 @@
+install_python3_pip:
+  pkg.installed:
+    - pkgs:
+        - python3
+        - python3-pip
+    - refresh: True
+    - require:
+      - pkg: install_python_3
+
+create_mitx_directory:
+  file.directory:
+    - name: /mitx
+
+clone_mitx_etl_repo:
+  git.latest:
+    - name: https://github.com/mitodl/odl-etl
+    - target: /mitx
+    - force_clone: True
+    - require:
+      - pkg: install_python_3
+
+install_mitx_residential_etl_requirements:
+  virtualenv.managed:
+    - name: /mitx/mitx_etl
+    - system_site_packages: False
+    - requirements: /mitx/requirements.txt
+    - env_vars:
+        PATH_VAR: '/usr/local/bin/pip3'
+    - require:
+      - git: clone_mitx_etl_repo
+      - pkg: install_python3_pip
+
+{% for key, value in salt.pillar.get('edx:mitx_etl:mitx_residential_etl:settings', {}).items() %}
+mitx_residential_etl_config:
```

Comment:
There's no need to use a for loop here. You can just pass the entire settings object through the JSON filter to render it out as a file. This will end up overwriting the file multiple times with each value.

Reply:
Removed for loop